### PR TITLE
Issue 7

### PR DESCRIPTION
**Pull request by krijnr**

### CODE REVIEW

**Comment by krijnr** [comment]
[Issue 7: Delete Image](https://trello.com/c/kf89Cy9T/331-issue-7-delete-image)

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -97,4 +97,11 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
       assert_redirected_to images_url
     end
   end
+
+  test 'should handle deleting non-existing image' do
+    assert_no_difference 'Image.count' do
+      delete image_path(-1)
+      assert_equal 'Image not found', flash[:alert]
```
Comment (infiniteluke) [inline test/controllers/images_controller_test.rb:104]: redirect can happen here too. we could assert here

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -86,4 +86,15 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
       assert_equal images_path(tag_filter: 'tag_b'), tags[0][:href]
     end
   end
+
+  test 'should delete an image' do
+    Image.create! path: 'https://www.demo.com/image3.jpg'
+    @image = Image.create! path: 'https://www.demo.com/image4.jpg', tag_list: 'tag_c, tag_d'
+    Image.create! path: 'https://www.demo.com/image5.jpg'
+    assert_difference 'Image.count', -1 do
```
Comment (eric199wen) [inline test/controllers/images_controller_test.rb:94]: To write clean code, we normally like to separate a test suite into three part, first is data setup, second is calling the action you want to test, third is assertion.

Comment (eric199wen) [reply test/controllers/images_controller_test.rb:94]: We could also assert the `@image` can no longer be found.

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (krijnr) [reply test/controllers/images_controller_test.rb:104]: got it. added check for redirect here

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (krijnr) [reply test/controllers/images_controller_test.rb:99]: separated the setup, action and assertion

**Pull request merged**